### PR TITLE
Redesign login and register / refactor create user

### PR DESCRIPTION
**Pull request by dennisreimann**
Addresses #1710. While working on this I came across two more things I chose to refactor – want to get your opinion on this as these things are a bit more involved and might need some more work. So I'll start with the refactorings:

1. Unified the welcome and error layouts to a single LayoutSimple: They were almost identical and hence there was a lot of duplicated code I decided to dry down.
2. Dissecting the register (user facing) and create user (admin area) workflows, giving them individual actions and views. This part might need more work, I just wanted to get it done as quickly as possible, because there might have objections here. IMHO it makes sense to split them up, because these are two separate use cases – which was hinted at with variables like `returnUrl`, `useBasicLayout` and `logon`. But I don't know the reasoning behind this and maybe you think it's better to keep it that way.

Alright, with that out of the way, here's what it looks like now.

High-level changes as proposed in the issue:

- One column layout to provide a clear structure 
- Register option is now a secondary button (might make sense for login on register page too)
- Make Tor-Link less prominent ([reasoning](https://github.com/btcpayserver/btcpayserver/issues/1710#issuecomment-652969599))
- Keep supporters on the page but move them to the bottom

## Mobile
mobile

## Desktop

desktop

## Admin: New user

create-user

new-user

## Error Pages

 The error pages still look the same as before, hence I'm leaving them out here.

### CODE REVIEW

**Comment by Eskyee** [comment]
Looking good 🔥

**Comment by pavlenex** [comment]
Nice progress. 
Supporters are still below the fold for me.
I suggest we also wait for feedback from designers in #1712 before we decide to further implement this.

File: BTCPayServer/Controllers/ServerController.cs [hunk 1]
```diff
@@ -303,6 +305,40 @@ public new async Task<IActionResult> User(string userId, UserViewModel viewModel
             return RedirectToAction(nameof(User), new { userId = userId });
         }
 
+        [Route("server/users/new")]
+        [HttpGet]
+        public IActionResult NewUser()
```
Comment (Kukks) [inline BTCPayServer/Controllers/ServerController.cs:310]: You could just name this the same as the post action `CreateUser` (and also keep the same route)

Comment (dennisreimann) [reply BTCPayServer/Controllers/ServerController.cs]: Thanks, done in 0235624c6424ec281a7e8970e84381f9e753dcf4.

**Comment by dennisreimann** [comment]
Rebased and incorporated the feeedback. 

- Login / Sign Up are now secondary buttons below the form
- Tor URL is now a button, which should be familiar to Tor users
- As we cannot open the onion URL directly in Tor browseer, I'bve made the button copy the URL to the clipboard
- Decreased the general page margin a bit, so that supporters should be visible above the fold

register-dark

login-light

login-dark

**Comment by pavlenex** [comment]
@dennisreimann Not a big fan of that Tor button, it looks like a main CTA. I again suggest waiting for feedback in #1710 regarding login page before proceeding further. We'd have to swap that font for example.  Then `BTCPAYSERVER SUPPORTERS` 😠 at the bottom should be changed. No need to have 2 links in form of tool tip, etc.

**Comment by dennisreimann** [comment]
After sleeping over it I agree regarind the position. Curious for the design feeeback in the issue, so that wee can finish this up :)

**Comment by pavlenex** [comment]
It'll take some time to get the ball rolling in there as there's only @dstrukt working on it. Will try to get him some help.

**Comment by dstrukt** [comment]
I agree @dennisreimann about the tor button feeling like a prominent CTA, which is why I opted for the icon + text to push it back in importance.

**Comment by bolatovumar** [comment]
Are we limiting the number of sponsors that can appear on the login page at once to six?

**Comment by dstrukt** [comment]
I think we should, but have them cycle through (assuming we have as many as 12 or 18 at any given time).

I'll provide an animation.

**Comment by pavlenex** [comment]
> 
> 
> Are we limiting the number of sponsors that can appear on the login page at once to six?

Only a certain tier of supporters is displayed on the front page. So we're not limit it.

**Comment by dennisreimann** [comment]
> I think we should, but have them cycle through (assuming we have as many as 12 or 18 at any given time).

Yeah, maybe, but let's not overcomplicate it and we can do this in another PR. Let's get this one finished up and improve it once there's a need for this.

**Comment by dstrukt** [comment]
> > I think we should, but have them cycle through (assuming we have as many as 12 or 18 at any given time).
> 
> Yeah, maybe, but let's not overcomplicate it and we can do this in another PR. Let's get this one finished up and improve it once there's a need for this.

Agreed, it's a scaling solution, if there's no need at this moment, I support this course of action.

**Comment by dennisreimann** [comment]
@NicolasDorier this isn't completely finished yet, but I'll incorporate the feedback in another PR.

**Comment by NicolasDorier** [comment]
1. Kukks approved
2. No WIP
3. Awesome
4. No @pavlenex bitching
5. Tests pass

Instant merge!

**Comment by pavlenex** [comment]
Kukks's approved it code-wise we were in the middle of discussing the design and UI.

**Comment by NicolasDorier** [comment]
Sorry boss! Github should not make the green button so enticing.

**Comment by Kukks** [comment]
Maybe we should start using the draft PR option more :P

**Pull request merged**